### PR TITLE
removed redundant pointer to firmware's s3 bucket and prefix from the…

### PR DESCRIPTION
**Pull request by ekalosak**
… requirements.yaml.m4; broke the s3 path specification out into bucket, prefix, and version in fetch_firmware.sh (and fetch_firmware.sh.m4)

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018121420-2-g305264a7

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=815770)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail"
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2018121420-2-g305264a7&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-post&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2018121420-2-g305264a7&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018121420-2-g305264a7
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018121420-2-g305264a7)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018121420-2-g305264a7/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018121420-2-g305264a7

**Comment by silverjam** [comment]
Is there a corresponding piksi-releases change for this?

**Comment by axlan** [comment]
It looks like we're still missing a layer here. When we talked about this, I didn't realized that there was a `fetch_firmware.sh.m4` in addition to the `requirements.yaml.m4` . M4 is a tool for doing substitutions in a file for example https://github.com/swift-nav/piksi_buildroot/pull/681 mentions:
```
m4 -DM4_BUCKET=swiftnav-artifacts -DM4_FW_VERSION=v1.5.0-develop-2018062919 -DM4_NAP_VERSION=v1.5.0-develop-2018062919 fetch_firmware.sh.m4 > fetch_firmware.sh
```
to generate a new `fetch_firmware.sh`

So it looks like the bucket is already specified for fetch_firmware.sh.m4, but for some reason isn't also a variable in requirements.yaml.m4 . 

You shouldn't need to change `fetch_firmware.sh.m4` or `fetch_firmware.sh.m4` , but we do need to figure out what's generating these scripts and update `requirements.yaml.m4`

@silverjam

**Comment by silverjam** [comment]
The release tool `piksi-releases` is generating `fetch_firmware.sh` from `fetch_firmware.sh.m4` (it is good practice to just update both of them), and the `requirements.yaml.m4` is generated by `fetch_firmware.sh` when invoked with `GENERATE_REQUIREMENTS=y` is in the environment (sorry, this was/is a gross hack that should be cleaned up).

The `requirement.yaml` is generated after a build succeeds on Travis/Jenkins and is then uploaded for HITL to consume.

If parameters are added to `requirements.yaml.m4` then `fetch_firmware.sh` / `fetch_firmware.sh.m4` needs to be updated so that they'll get substituted by m4.  See https://github.com/swift-nav/piksi_buildroot/blob/master/fetch_firmware.sh.m4#L76.

File: fetch_firmware.sh [hunk 1]
```diff
@@ -32,7 +32,9 @@ FW_VERSION=${1:-v2.1.0-develop-2018121420}
 NAP_VERSION=${2:-v2.1.0-develop-2018121420}
 
 CCACHE_S3_PATH=s3://swiftnav-artifacts/piksi_buildroot/$BR_VERSION
-FW_S3_PATH=s3://swiftnav-artifacts/piksi_firmware_private/$FW_VERSION/v3
+FW_S3_BUCKET=swiftnav-artifacts
+FW_S3_PREFIX=piksi_firmware_private
+FW_S3_PATH=s3://$FW_S3_BUCKET/$FW_S3_PREFIX/$FW_VERSION/v3
 NAP_S3_PATH=s3://swiftnav-artifacts/piksi_fpga/$NAP_VERSION
 NAP_S3_PATH_PROD=s3://swiftnav-artifacts/piksi_fpga/$NAP_VERSION
```
Comment (mfine) [inline fetch_firmware.sh]: Can you add variables for the NAP too while you're in here?

Comment (mfine) [reply fetch_firmware.sh]: Was this actually generated by M4? It should read:

```
FW_S3_BUCKET=${3:-swiftnav-artifacts}
FW_S3_PREFIX=${4:-piksi_firmware_private}
```

File: fetch_firmware.sh.m4 [hunk 2]
```diff
@@ -32,7 +32,9 @@ FW_VERSION=${1:-M4_FW_VERSION}
 NAP_VERSION=${2:-M4_NAP_VERSION}
 
 CCACHE_S3_PATH=s3://swiftnav-artifacts/piksi_buildroot/$BR_VERSION
-FW_S3_PATH=s3://M4_BUCKET/piksi_firmware_private/$FW_VERSION/v3
+FW_S3_BUCKET=${3:-M4_FW_S3_BUCKET}
+FW_S3_PREFIX=${4:-M4_FW_S3_PREFIX}
+FW_S3_PATH=s3://M4_BUCKET/$FW_S3_PREFIX/$FW_VERSION/v3
```
Comment (mfine) [inline fetch_firmware.sh.m4]: Can this be moved up with the other passed in substitutions?

```
FW_VERSION=${1:-M4_FW_VERSION}
NAP_VERSION=${2:-M4_NAP_VERSION}
FW_S3_BUCKET=${3:-M4_FW_S3_BUCKET}
FW_S3_PREFIX=${4:-M4_FW_S3_PREFIX}
```

File: fetch_firmware.sh (hunk 1 above)
Comment (ekalosak) [reply fetch_firmware.sh]: It was not generated by M4 - I'll work on this. Thanks for the feedback.

Comment (ekalosak) [reply fetch_firmware.sh]: Suggested change implemented in commit 876cfd03311dfc201a068d41d2b5cfa2bbecb1a9

File: fetch_firmware.sh.m4 (hunk 2 above)
Comment (ekalosak) [reply fetch_firmware.sh.m4]: Suggested change implemented in commit 876cfd03311dfc201a068d41d2b5cfa2bbecb1a9

Comment (silverjam) [reply fetch_firmware.sh.m4]: I'm not able to find this commit, is it in this repo?

**Pull request closed without merging**